src/MemoryRouter.test.tsx

```diff
@@ -62,6 +68,19 @@ describe("MemoryRouter", () => {
           });
         });
 
+        it("should trigger only hashEvents on a hash change event", async () => {
```

Comment:
I would love a few more test cases to ensure we capture the Next behavior.

Reply:
I added unit tests for (hopefully) all cases